cpu/stm32_common/periph/eeprom.c

```diff
@@ -30,25 +33,155 @@
 
 extern void _lock(void);
 extern void _unlock(void);
+extern void _wait_for_pending_operations(void);
 
 #ifndef EEPROM_START_ADDR
 #error "periph/eeprom: EEPROM_START_ADDR is not defined"
 #endif
 
-uint8_t eeprom_read_byte(uint32_t pos)
+static void _eeprom_write_byte(uint32_t pos, uint8_t byte)
 {
     assert(pos < EEPROM_SIZE);
+    
+    _wait_for_pending_operations();
+    *(__IO uint8_t *)(EEPROM_START_ADDR + pos) = byte;
+}
+
+static void _eeprom_write_word(uint32_t pos, uint32_t word)
+{
+    assert(pos < EEPROM_SIZE - sizeof(uint32_t));
+    
+    _wait_for_pending_operations();
+    *(__IO uint32_t *)(EEPROM_START_ADDR + pos) = word;
+}
 
-    DEBUG("Reading data from EEPROM at pos %" PRIu32 "\n", pos);
+static uint8_t _eeprom_read_byte(uint32_t pos)
+{
+    assert(pos < EEPROM_SIZE);
+    
+    _wait_for_pending_operations();
     return *(uint8_t *)(EEPROM_START_ADDR + pos);
 }
 
+static uint32_t _eeprom_read_word(uint32_t pos)
+{
+    assert(pos < EEPROM_SIZE - sizeof(uint32_t));
+    
+    _wait_for_pending_operations();
+    return *(uint32_t *)(EEPROM_START_ADDR + pos);
+}
+
 void eeprom_write_byte(uint32_t pos, uint8_t data)
 {
     assert(pos < EEPROM_SIZE);
 
-    DEBUG("Writing data '%c' to EEPROM at pos %" PRIu32 "\n", data, pos);
+    DEBUG("Writing data '%c' to EEPROM at pos %lu\n", data, pos);
     _unlock();
-    *(uint8_t *)(EEPROM_START_ADDR + pos) = data;
+    _eeprom_write_byte(pos, data);
     _lock();
 }
+
+size_t eeprom_write(uint32_t pos, const uint8_t *data, size_t len)
+{
+    assert(pos+len < EEPROM_SIZE);
+
+    DEBUG("[EEPROM] write %d bytes\n", len);
+    _unlock();
+    uint32_t i;
+    uint32_t bytes = 0;
+    /* write first bytes in case of unaligned access */
+    uint32_t shift = pos & 0x3;
+    if (shift > 0) {
+        for (i = shift; i < 4; i++) {
+            if (data)
+                _eeprom_write_byte(pos + bytes, data[bytes]);
+            else
+                _eeprom_write_byte(pos + bytes, 0);
+            bytes++;
+        }
+    }
+    
+    /* write by words to speed up the process */
+    uint32_t words = (len - bytes)/4;
```

Comment:
In general, we put spaces around operators (although it's explicitly stated in [RIOT coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions))